components/helm-broker/docs/proposals/addon-git-support.md

```diff
@@ -0,0 +1,163 @@
+# Support for loading Addons from Git repository
+
+Created on 2019-07-24 by Mateusz Szostok ([@mszostok](https://github.com/mszostok)).
+
+This document describes options for extending the ClusterAddonsConfiguration/AddonsConfiguration for supporting loading Addons from the Git repository.
+
+## Motivation
+
+Helm Broker fetches addons listed in the `index.yaml` file from remote HTTPS servers. Only zipped bundles can be fetched. This solution generates problems for end-user as it's not easy to always have dedicated HTTPS server which serves zipped bundles. A most common scenario is to have a git repository with unzipped bundles and index YAMLs. This document describes possible and accepted solutions for achieving that goal.
```

Comment:
Please mentioned at the end, that this syntax in addition opens a doors to other types of storage, like s3, mercurial etc.